src/request/getGrantFromRedirectUrl.ts

```diff
@@ -0,0 +1,83 @@
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { UrlString } from "@inrupt/solid-client";
+import {
+  isVerifiableCredential,
+  VerifiableCredential,
+} from "@inrupt/solid-client-vc";
+import {
+  GRANT_VC_PARAM_NAME,
+  GRANT_VC_URL_PARAM_NAME,
+} from "../manage/redirectToRequestor";
+import { getSessionFetch } from "../util/getSessionFetch";
+
+/**
+ * Get the Access Grant out of the incoming redirect from the Access Management app.
+ *
+ * @param redirectUrl The URL the user has been redirected to from the access
+ * management app.
+ * @param option Optional properties to customise the behaviour:
+ * - fetch: an authenticated fetch function. If not provided, the default session
+ * from @inrupt/solid-client-authn-browser will be used if available.
+ * @returns An Access Grant
+ */
+export async function getGrantFromRedirectUrl(
+  redirectUrl: UrlString,
+  option: { fetch?: typeof fetch } = {}
+): Promise<VerifiableCredential> {
+  const redirectUrlObj = new URL(redirectUrl);
+  const authFetch = option.fetch ?? (await getSessionFetch(option));
+
+  const accessGrantValue = redirectUrlObj.searchParams.get(GRANT_VC_PARAM_NAME);
+  const accessGrantIri = redirectUrlObj.searchParams.get(
+    GRANT_VC_URL_PARAM_NAME
+  );
+  if (accessGrantIri === null && accessGrantValue === null) {
+    throw new Error(
+      `The provided redirect URL [${redirectUrl}] is missing the expected [${GRANT_VC_URL_PARAM_NAME}] query parameter`
+    );
+  }
+  const accessGrant = accessGrantValue
+    ? JSON.parse(atob(accessGrantValue))
+    : await authFetch(accessGrantIri as string).then(async (response) => {
+        if (!response.ok) {
+          throw new Error(
+            `Fetching the Access Grant [${accessGrantIri}] failed: ${response.status} ${response.statusText}`
+          );
+        }
+        try {
+          return await response.json();
+        } catch (e) {
+          throw new Error(
+            `Parsing the Access Grant [${accessGrantIri}] as JSON failed: ${e.toString()}`
+          );
+        }
+      });
+  if (!isVerifiableCredential(accessGrant)) {
+    throw new Error(
+      `The value received from [${accessGrantIri}] is not a Verifiable Credential`
+    );
+  }
+  return accessGrant;
```

Comment:
This should be extracted as a `getVerifiableCredential` function call (and used by both getAccessGrant and getAccessRequest).

Reply:
See https://github.com/inrupt/solid-client-vc-js/pull/284